app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java

```diff
@@ -0,0 +1,306 @@
+            //get name without extension
+            String fileName = Utils.getFilePrefix(filePath);
+
+            //check if the file is a test
+            if(fileName.contains(PHOTO_BASE_NAME)) {
+
+                String fileExtension = Utils.getFileExtension(filePath);
+
+                //check if image extension is available
+                if(Arrays.asList(IMAGE_EXTENSIONS).contains(fileExtension)) {
+
+                    Bitmap photoBitmap = Utils.loadBitmapFromFile(filePath);
+
+                    //Each photo has a description.txt with the same filename - so when an image is found we know the description filename
+                    String photoDesc= Utils.getTextFromFile(dirPath + "/" + fileName + ".txt");
+
+                    JSONObject jsonPhotoDescription = null;
+                    try {
+                        //create test element giving filename, description and bitmap
+                        jsonPhotoDescription = new JSONObject(photoDesc);
+                    } catch(JSONException e) {
+                        e.printStackTrace();
+                        Log.e(TAG, "PhotoTester constructor -> Error decoding JSON with name " + fileName);
+                    }
+                    if(jsonPhotoDescription != null) {
+                        TestElement originalTest = new TestElement(photoBitmap, jsonPhotoDescription, fileName);
+
+                        //associate the relative bitmap to each alteration of the original test if there is any
+                        String[] alterationsFilenames = originalTest.getAlterationsNames();
+                        if(alterationsFilenames != null) {
+                            for(String alterationFilename : alterationsFilenames) {
+
+                                String alterationBitmapPath = dirPath + "/" + alterationFilename;
+                                Bitmap alterationBitmap = Utils.loadBitmapFromFile(alterationBitmapPath);
+                                //Utils.loadBitmapFromFile may return null
+                                if(alterationBitmap != null)
+                                    originalTest.setAlterationBitmap(alterationFilename, alterationBitmap);
+                                else
+                                    Log.i(TAG, "alteration image with name " + alterationFilename + " not found")
+                            }
+                        }
+                        testElements.add(originalTest);
+                    }
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Elaborate tests using threads, stores a report in JSON format to testReport.txt inside the directory given on construction
+     * @return String in JSON format with the test's report, each object is a single test named with the filename and contains:
+     * ingredients, tags, notes, original photo name, recognized text, confidence and alterations (if any), each alteration
+     * contains tags, notes, recognized text, confidence
```

Comment:
I did not really understand what is "confidence", i think it could be more explanatory

Reply:
Thank you for reporting that. Confidence is a float that describes the correspondence between the recognized text from the OCR and the real one.